src/Demos/HSP.lagda

```diff
@@ -204,49 +176,31 @@ module _ {A : Type α }{B : A → Type β} where
-relation manifesting the notion of equality of elements of the set. We
-often take this equivalence for granted or view it as self-evident; rarely do we
-take pains to define it explicitly. While well-suited to informal
-mathematics, this approach is inadequate for formal, machine-checked proofs.
-
-The \agdaalgebras library was first developed without setoids, relying exclusively
-on the inductive equality type \ad{\au{}≡\au{}}, defined in \am{Agda.Builtin.Equality},
+A \defn{setoid} is a pair of a type \ab A and
+an equivalence relation \af{≈} on \ab A.  They are useful to represent a
+carrier with a ``local'' notion of equivalence, instead of always relying on
+the global one as is usually done in set theory. Dealing with setoids has
+somewhat of a bad reputation for being overly tedious. We have not found
+this to be an issue.
+
+The \agdaalgebras library was first developed without setoids, relying on
+propositional equality \ad{\au{}≡\au{}} instead,
 along with some experimental, domain-specific types for equivalence classes, quotients, etc.
-One consequence of this design decision was that the formalization of many
-theorems required postulating function extensionality, an axiom that is known to be neither provable
-nor refutable in pure Martin-Löf type theory.\footnote{See the section
-\href{https://www.cs.bham.ac.uk/~mhe/HoTT-UF-in-Agda-Lecture-Notes/HoTT-UF-Agda.html\#funextfromua}
-{\textit{Function extensionality from univalence}} in~\cite{MHE, MHE:2019}.}
-
-In contrast, our current approach using setoids makes the equality relation
-of a given type explicit.  A primary motivation for this choice is to avoid the need for
-additional axioms and to make it clearer that the formal proofs in the \agdaalgebras
-library are fully \emph{constructive} (as defined in~\cite{nlab:constructive_mathematics})
-and confined to \emph{Martin-Löf dependent type theory} (as defined in~\cite{nlab:martin-loef_dependent_type_theory}).
-In particular, we make no appeals to classical axioms like Choice or Excluded Middle, nor
-do we postulate function extensionality at any point in the present work.\footnote{The \defn{function extensionality
-axiom} asserts that two point-wise equal functions are equal. There remain some modules in
-\agdaalgebras that postulate this axiom, but we don't use them here.}
-We are confident that the \agdaalgebras library is now fully constructive and free from any hidden
-assumptions or inconsistencies that could be used to fool a type-checker.\footnote{As of 7 Dec 2021, the latest version of \agdaalgebras is 2.0.1; see~\cite{ualib_v2.0.1}.}
+This furthermore required postulating function extensionality%
+\footnote{An axiom that asserts that two point-wise equal functions are equal.}, which is
+known to be independent from \mltt~\cite{MHE, MHE:2019}. We felt that this was
+an unsatisfactory state of affairs, as we were curious to see if the theorems
```

Comment:
Nice revisions overall, but instead of "...as we were curious to see if the theorems held...", I would prefer, "...as our aim is to show that the theorems hold..."